### PR TITLE
Patch the `BayesGMMTransformer`

### LINKED ISSUE #348: Patching the `BayesGMMTransformer`
There are a few improvements that need to be made in the `BayesGMMTransformer`:

1. The variable `_valid_component_indicator` should actually actually be public, so the `_` should be removed.
2. The integration tests `test_multiple_components` and `test_nulls` fail randomly, so a random seed should be set.
3. Add an integration test where all the values are nulls.

### PR DESCRIPTION
**Pull request by fealho**
Resolve #348.

### CODE REVIEW

File: tests/integration/transformers/test_numerical.py [hunk 1]
```diff
@@ -180,6 +181,21 @@ def test_nulls(self):
...
+        data = pd.DataFrame([np.nan, None] * 50, columns=['col'])
+        bgmm_transformer = BayesGMMTransformer()
+        bgmm_transformer.fit(data, list(data.columns))
+        transformed = bgmm_transformer.transform(data)
+
+        assert isinstance(transformed, pd.DataFrame)
+        assert transformed.shape == (100, 3)
+        assert all(x == 0.0 for x in transformed['col.normalized'])
+        assert all(x == 0.0 for x in transformed['col.component'])
+        assert all(x == 1.0 for x in transformed['col.is_null'])
```
Comment (amontanez24) [inline tests/integration/transformers/test_numerical.py:194]: Could you make an expected DataFrame like
```
expected_transformed = pd.DataFrame({
    'col.normalized': [0] * 100,
    'col.component': [0] * 100,
    'col.is_null': [1.0] * 100
})
```
and just compare that transformed is equal to it?

File: tests/integration/transformers/test_numerical.py [hunk 2]
```diff
@@ -180,6 +181,21 @@ def test_nulls(self):
...
+
+        reverse = bgmm_transformer.reverse_transform(transformed)
+        np.testing.assert_array_almost_equal(reverse, data)
```
Comment (amontanez24) [inline tests/integration/transformers/test_numerical.py:197]: Is it possible for this to ever fail? If so, should we use a random seed here too?

**Review by amontanez24 — APPROVED** [review]
A couple minor comments, but besides that it looks good to me!

File: tests/integration/transformers/test_numerical.py (hunk 2 above)
Comment (fealho) [reply tests/integration/transformers/test_numerical.py:197]: I don't think so (at least it never happened to me), but i'll add it just in case.

**Review by pvk-developer — APPROVED** [review]
LGTM!

**Pull request merged**